src/modules/forum/sagas.ts

```diff
@@ -2,21 +2,51 @@ import { takeEvery, call, put } from 'redux-saga/effects'
 import { builder } from 'lib/api/builder'
 import {
   createCollectionForumPostFailure,
+  CreateCollectionForumPostFailureAction,
+  createCollectionForumPostRequest,
   CreateCollectionForumPostRequestAction,
   createCollectionForumPostSuccess,
+  CREATE_COLLECTION_FORUM_POST_FAILURE,
   CREATE_COLLECTION_FORUM_POST_REQUEST
 } from './actions'
+import { delay } from 'dcl-catalyst-commons'
+
+const RETRY_DELAY = '5000'
 
 export function* forumSaga() {
   yield takeEvery(CREATE_COLLECTION_FORUM_POST_REQUEST, handleCreateForumPostRequest)
+  yield takeEvery(CREATE_COLLECTION_FORUM_POST_FAILURE, handleCreateForumPostFailure)
 }
 
 function* handleCreateForumPostRequest(action: CreateCollectionForumPostRequestAction) {
   const { collection, forumPost } = action.payload
   try {
-    const forumLink: string = yield call(() => builder.createCollectionForumPost(collection, forumPost))
+    console.log('Creating forum post request')
+    const forumLink: string = yield call([builder, 'createCollectionForumPost'], collection, forumPost)
     yield put(createCollectionForumPostSuccess(collection, forumLink))
   } catch (error) {
-    yield put(createCollectionForumPostFailure(collection, error.message))
+    console.log('Failed forum post request', error)
+    const forumPostAlreadyExists = error?.response?.status === 409
+    const forumLink = error?.response?.data?.data?.forum_link
```

Comment:
I think we can maybe take advantage of the fact that we're using Axios behind the scenes and type this a bit better. We already have a private method on builder.ts which does:

```ts
private isAxiosError(error: any): error is AxiosError {
```

so if we make it public and use it here we could

```ts
if (builder.isAxiosError(error)) {
  // error correctly typed here
}
```

to maybe avoid a few `.?`

Reply:
The thing is that the only thing that I can remove is the optional chaining operator of error, that is the one that checks that error isn't null. That's because the first property, response, can be undefined in cases where the exception comes from Axios and in case where it doesn't. [Here's an example](https://github.com/axios/axios#handling-errors) on handling error that come from Axios.
I've removed that chaining operator to be a bit more clear on what we can receive.